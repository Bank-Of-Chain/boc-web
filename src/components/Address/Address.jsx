import React from 'react'
import PropTypes from 'prop-types'
import useLookupAddress from '@/hooks/useLookupAddress'

// === Utils === //
import isEmpty from 'lodash-es/isEmpty'

const Address = props => {
  const { ensProvider, address = '' } = props
  if (isEmpty(address)) {
    return ''
  }
  const ens = useLookupAddress(ensProvider, address)

  let displayAddress = address.substr(0, 6)

  if (ens && ens.indexOf('0x') < 0) {
    displayAddress = ens
  } else if (props.size === 'short') {
    displayAddress += '...' + address.substr(-4)
  } else if (props.size === 'long') {
    displayAddress = address
  }
  return <span>{displayAddress}</span>
}

Address.propTypes = {
  address: PropTypes.string.isRequired
}

export default Address
