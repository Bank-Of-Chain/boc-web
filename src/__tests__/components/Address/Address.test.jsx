import React from 'react'
import { render } from '@testing-library/react'
import Address from '@/components/Address/Address'
import { describe, expect, it } from 'vitest'

describe('xxxxxxxx', () => {
  it('Address Component Render', () => {
    const { asFragment } = render(<Address address="0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266" size="long" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
