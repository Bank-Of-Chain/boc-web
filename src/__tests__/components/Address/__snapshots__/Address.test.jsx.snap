// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Address Component Render 1`] = `
<DocumentFragment>
  <span>
    0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
  </span>
</DocumentFragment>
`;
